### PR TITLE
Fix race condition in raft test.

### PR DESCRIPTION
**Pull request by ExternalReality**
## Description of change

This PR fixes a race condition in a Raft test. 

Why is this change needed?

The change is meant to help make the test suite more resilient against intermittent failures.

## QA steps

How do we verify that the change works?

`go test -race ./worker/raft/...`

You should see neither tests fail nor a race detected.

## Documentation changes

Does it affect current user workflow? CLI? API?

N/A

## Bug reference

https://bugs.launchpad.net/juju/+bug/1749005

### CODE REVIEW

**Comment by ExternalReality** [comment]
[Fix raft cluster test race condition](https://trello.com/c/szVlvNJ5/72-fix-raft-cluster-test-race-condition)

**Comment by ExternalReality** [comment]
!!build!!

File: worker/raft/raftclusterer/worker_test.go [hunk 1]
```diff
@@ -286,3 +253,15 @@ func (s *WorkerSuite) publishDetails(c *gc.C, details apiserver.Details) {
 		c.Fatal("timed out waiting for details to be received")
 	}
 }
+
+// Connect  all of the provided transport bidirectionally.
+func connectTransports(transports []raft.LoopbackTransport) {
```
Comment (jameinel) [inline worker/raft/raftclusterer/worker_test.go]: I think if you change this to:
func connectTransports(transports []raft.LoopbackTransport...)

you can then call it with
connectTransports(t1, t2, t3)

and not have to cast them into a slice in the test code.

**Comment by ExternalReality** [comment]
$$MERGE$$

**Pull request merged**